lib/rubocop/cop/sorbet/forbid_include_const_literal.rb

```diff
@@ -26,41 +26,44 @@ module Sorbet
       # ```ruby
       # include Polaris::Engine.helpers
       # ```
-      class ForbidIncludeConstLiteral < RuboCop::Cop::Cop # rubocop:todo InternalAffairs/InheritDeprecatedCopClass
-        MSG = "Includes must only contain constant literals"
+      class ForbidIncludeConstLiteral < RuboCop::Cop::Base
+        extend AutoCorrector
 
-        attr_accessor :used_names
+        MSG = "`%<inclusion_method>s` must only be used with constant literals as arguments"
+        RESTRICT_ON_SEND = [:include, :extend, :prepend].freeze
 
-        # @!method not_lit_const_include?(node)
-        def_node_matcher :not_lit_const_include?, <<-PATTERN
-          (send nil? {:include :extend :prepend}
-            $_
-          )
+        # @!method dynamic_inclusion?(node)
+        def_node_matcher :dynamic_inclusion?, <<~PATTERN
+          (send $_ ${:include :extend :prepend} $#neither_const_nor_self?)
         PATTERN
 
-        def initialize(*)
-          super
-          self.used_names = Set.new
-        end
+        def_node_matcher :t_unsafe?, <<~PATTERN
+          (send (const {nil? cbase} :T) :unsafe _)
+        PATTERN
 
         def on_send(node)
-          return unless not_lit_const_include?(node) do |send_argument|
-            ![:const, :self].include?(send_argument.type)
+          dynamic_inclusion?(node) do |receiver, inclusion_method, included|
+            return if t_unsafe?(receiver)
+            return unless within_onymous_module?(node)
+
+            add_offense(node, message: format(MSG, inclusion_method: inclusion_method)) do |corrector|
+              explicit_receiver = receiver&.source || 'self'
```

Comment:
This seems to be a bug-fix so I think we have no test cases for the explicit receiver not being `self`. Can we add one or two cases for that as well?

Reply:
I've added tests, although I should note that they look like

```ruby
module M
  M.include(dynamic)
end
```

which this implementation detects, not

```ruby
M.include(dynamic)
```

which this implementation does _not_ detect. Do we want to do that? I'm unsure about false positives.